src/software/jetson_nano/services/motor.cpp

```diff
@@ -624,6 +628,40 @@ void MotorService::spiTransfer(int fd, uint8_t const* tx, uint8_t const* rx, uns
 }
 
 
+void MotorService::readThenWriteSpiTransfer(int fd, const uint8_t* read_tx,
+                                            const uint8_t* write_tx,
+                                            const uint8_t* read_rx, uint32_t spi_speed)
+{
+    int ret1, ret2;
+
+    uint8_t write_rx[5] = {0};
+
+    struct spi_ioc_transfer tr[2];
+    memset(tr, 0, sizeof(tr));
+
+    tr[0].tx_buf        = (unsigned long)read_tx;
+    tr[0].rx_buf        = (unsigned long)read_rx;
+    tr[0].len           = 5;
```

Comment:
minor nit: Could we add a constant for the `5`s in this method (assuming they all refer to the same thing). And could you maybe add a comment of why the size is 5 or where it's from? 
Edit: Seems like it might be based on the trinamic transaction msg side according to `MotorService::tmc4671ReadThenWriteValue`. I wonder if this method should also have a `len` argument similar to `MotorService::spiTransfer`

Reply:
ya it corresponds to a size of command message to a trinamic packet

Reply:
fixed